SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/WorkerCoordinator/AbstractWorkerCoordinator.cs

```diff
@@ -51,20 +60,28 @@ protected Process CreateSimulatedPlayerProcess(string fileName, string args)
 
         /// <summary>
         /// Blocks until all active simulated player processes have exited.
+        /// Restarts failed simulated player processes.
         /// Will only wait for processes started through CreateSimulatedPlayerProcess().
         /// </summary>
         protected void WaitForPlayersToExit()
         {
-            while (ActiveProcesses.Count > 0)
+            while (true)
             {
-                try
+                var finishedProcesses = ActiveProcesses.Where(process => process.HasExited).ToList();
```

Comment:
Not too sure about `ToList` tbh - but at least we won't have any sudden changes in the states of these processes, should one of them fail between this and the next line

Reply:
What are you worried about?

Reply:
So initially I had something like this:
```
var finishedProcesses = ActiveProcesses.Where(process => process.HasExited);

var incorrectlyFinishedProcesses = finishedProcesses.Where(process => process.ExitCode != 0);

if (incorrectlyFinishedProcesses.Count() == 0 && finishedProcesses.Count() == ActiveProcesses.Count)
{
    return;
}

foreach (var process in incorrectlyFinishedProcesses)
{
    process.Start();
}
```
So `finishedProcesses` and `incorrectlyFinishedProcesses` are LINQ queries on top of `ActiveProcesses`. This means that every time I enumerate over them, the returned collection may be different, as the actual query is invoked. 

For the `if` I'm wary of the first operand `incorrectlyFinishedProcesses.Count() == 0` returning `true` (so no processes have crashed), but by the time the second part of the condition is invoked, every process may have crashed, so the WorkerCoordinator would exit.

In case of the `foreach` loop, this easily leads to `incorrectlyFinishedProcesses` being modified during the iteration (as `process.Start()` changes the HasExited and ExitCode of the app, or at least this is what I'd expect it to do). 

I'm not really that confident in my C# to say for sure that this won't cause any issues, and with `ToList()` freezing the collection this is of no concern.

Reply:
I am not C# expert either, but that seems like solid reasoning to me and a common pattern I see in c++ too (in the form of taking a copy of an array in c++s case). So lgtm